numpy/core/tests/test_shape_base.py

```diff
@@ -372,22 +375,71 @@ def test_stack():
                         stack, [np.arange(2), np.arange(3)])
 
 
+# See for more information on how to parametrize a whole class
+# https://docs.pytest.org/en/latest/example/parametrize.html#parametrizing-test-methods-through-per-class-configuration
+def pytest_generate_tests(metafunc):
+    # called once per each test function
+    if hasattr(metafunc.cls, 'params'):
+        arglist = metafunc.cls.params
+        argnames = sorted(arglist[0])
+        metafunc.parametrize(argnames,
+                             [[funcargs[name] for name in argnames]
+                              for funcargs in arglist])
```

Comment:
This seems like a lot of magic for something that can be achieved with a parametrized fixture

Reply:
Off the top of my head, I think something like:
```
class Tests:
    @pytest.fixture(params=[block1, block2, ...])
    def block(request): # maybe needs self, not sure
        return request.param

    def method(self, block): ...
```

Reply:
do you have a working example of a fixture i can build off?

Reply:
Does the one above not work?